src/Transport/AzureServiceBusTransport.cs

```diff
@@ -333,5 +333,14 @@ public IWebProxy WebProxy
 
         internal string FullyQualifiedNamespace { get; set; }
         internal TokenCredential TokenCredential { get; set; }
+
+        /// <summary>
+        /// Disable legacy headers for wire compatibility with NServiceBus.AzureServiceBus
+        /// </summary>
+        public void DisableLegacyHeaders()
```

Comment:
FYI when you look at the transport definition we opted for a more property-based model. Only the legacy shim extensions that make the transport compatible with `UseTransport<AzureServiceBusTransport>` uses methods.

Reply:
> FYI when you look at the transport definition we opted for a more property-based model. Only the legacy shim extensions that make the transport compatible with `UseTransport<AzureServiceBusTransport>` uses methods.

I opted for a method because of "Disable". Feels more natural as an "opt-out" API compared to doing:

```
transport.DisableLegacyHeaders = true
```

Alternative could  be:

```
transport.LegacyHeaders = false
```

Now that I think a bit longer about this API it likely would be better to mention the compabitibility in the name:

```
transport.DisableLegacyTransportCompatibility()
```

Reply:
Look at other transports

https://github.com/Particular/NServiceBus.AmazonSQS/blob/master/src/NServiceBus.Transport.SQS/Configure/SqsTransport.cs
https://github.com/Particular/NServiceBus.RabbitMQ/blob/master/src/NServiceBus.Transport.RabbitMQ/RabbitMQTransport.cs
https://github.com/Particular/NServiceBus.SqlServer/blob/master/src/NServiceBus.Transport.SqlServer/SqlServerTransport.cs
https://github.com/Particular/NServiceBus.SqlServer/blob/master/src/NServiceBus.Transport.PostgreSql/PostgreSqlTransport.cs
https://github.com/Particular/NServiceBus.Transport.Msmq/blob/master/src/NServiceBus.Transport.Msmq/MsmqTransport.cs

etc. They all use properties to express the simple configuration flags and name the properties according to the intent (some of the internal ones even contain `Disable`). The only different API is the one on Rabbit with the Add semantic where it makes sense even though it could also have been expressed with a more complex type on a property.

Opting for a different approach based on the above justification doesn't seem warranted because it seems to fall into the category of "I like it better this way".

Reply:
Furthermore, [considering this guidance](https://github.com/Particular/Platform/blob/main/documentation/transports.md) (copying and pasting for reference)

> The shim API methods are currently marked with a [[PreObsolete] attribute](https://github.com/Particular/Platform/blob/main/documentation/pre-obsolete-attribute.md) - this is in preparation to [making the API obsolete](https://github.com/Particular/Platform/blob/main/guidelines/breaking-changes.md#deprecating-an-api). However, **since the API is not yet marked as obsolete, and hence users are not seeing warnings about using this API, any new transport settings should always be added to both**. This will ensure that users of both implementation methods are able to make use of new features without rewriting their code.
> 
> Once we decide to retire the shim API and its methods are marked as obsolete, we can then stop modifying it and have new settings only added to the new API.

It means the missing shim API method would clash anyway.

Reply:
Why not `DoNotSendTransportEncodingHeaders`. The more general the name is, the more likely it will collapse with some other headers that we might need to deprecate. So, exposing a very concrete property for the specific header we are no longer sending seems to be better, no? Because the term "Legacy" is very broad and might chance over the multiple version of the transport we have to release until we can finally deprecate the setting, having such a broad term seems almost dangerous.

Reply:
That header is only used by ASBL. Users might otherwise not know why they could invoke that but that still is in the property docs. I like the naming compared to "Disabled" so adopted that.

Reply:
DoNotSendTransportEncodingHeaderForBackwardsCompatibilityWithPreviousTransportVersions? :dancing_men:

Reply:
I applied your suggestion